src/dbnode/storage/index/options.go

```diff
@@ -64,6 +64,9 @@ const (
 	aggregateResultsEntryArrayPoolSize        = 256
 	aggregateResultsEntryArrayPoolCapacity    = 256
 	aggregateResultsEntryArrayPoolMaxCapacity = 256 // Do not allow grows, since we know the size
+
+	// defaultResultsPerPermit sets the default index results that can be processed per permit acquired.
+	defaultResultsPerPermit = 10000
```

Comment:
not sure what a good default is. I'm hesitant to make the default really large or off, since we changed the access pattern for acquiring workers. now all blocks eagerly attempt to acquire workers in parallel, so without some kind of max, larger queries will dominate even more.